node/src/client/mod.rs

```diff
@@ -529,7 +536,14 @@ impl<N: Network, C: ConsensusStorage<N>> Client<N, C> {
                                 _node.propagate(Message::UnconfirmedTransaction(serialized), &[peer_ip]);
                             }
                             Err(error) => {
-                                debug!("Failed to verify the execution from peer_ip {peer_ip} - {error}");
+                                let error_string = error.to_string();
+                                if error_string.contains("global state root not found") {
```

Comment:
What about checking the state root inclusion before doing the `check_transaction_basic` calls?

Something like:
```
let state_roots = [
    transaction.execution().map(|t| t.global_state_root()),
    transaction.fee_transition().map(|t| t.global_state_root()),
]
    .into_iter()
    .flatten();

for state_root in state_roots {
    if !_node.ledger().contains_state_root(&state_root).unwrap_or(false) {
        debug!("Failed to find global state root for deployment from peer_ip {peer_ip}, propagating anyway");
        // Propagate the `UnconfirmedTransaction`.
        _node.propagate(Message::UnconfirmedTransaction(serialized), &[peer_ip]);
        break;
    }
}
```

This would now remove reliance on checking for the "global state root not found" errors.